plugins/modules/s3_bucket.py

```diff
@@ -2052,6 +2256,49 @@ def main():
             mutually_exclusive=[("days", "years")],
             required_one_of=[("days", "years")],
         ),
+        inventory=dict(
+            type="list",
+            elements="dict",
+            default=[],
```

Comment:
```suggestion
            default=[],
```

We should not be setting a default here: we should only be making changes to this setting if someone explicitly passes values.  Otherwise this module goes from being unable to manage the inventory settings to automatically nuking any that may have been configured outside of the module.